### ISSUE TITLE
Type generation: Query generated incorrectly when in a response

### ISSUE THREAD

**Issue by LMFinney** [nestjs/graphql#979]
## I'm submitting a...
<pre><code>
[ ] Regression 
[X] Bug report
[ ] Feature request
[ ] Documentation issue or request
[ ] Support request => Please do not submit support request here, instead post your question on Stack Overflow.
</code></pre>

## Current behavior
I ran `GraphQLDefinitionsFactory.generate()` to create the TypeScript interfaces for my schema, and the way Queries are handled is inconsistent when a Mutation's return type includes a Query. `Query` gets renamed to `IQuery` by `GraphQLAstExplorer`, but references to it do not get renamed.

This was [reported previously](https://github.com/nestjs/graphql/issues/123#issuecomment-463703501), but it seems that it was assumed to have been fixed.

## Expected behavior
If a root gets renamed, then references to the root should be renamed, too.

## Minimal reproduction of the problem with instructions
1. Use this input schema:
```
type Query {
  users(filter: String): String!
}

type Mutation {
  create(input: String!): CreateResponse!
}

type CreateResponse {
  data: String
  query: Query
}
```
2. Generate the output with this code:
```
import { GraphQLDefinitionsFactory } from '@nestjs/graphql';
import { join } from 'path';

const definitionsFactory = new GraphQLDefinitionsFactory();
definitionsFactory.generate({
  typePaths: ['./**/*.graphqls'],
  path: join(process.cwd(), 'output.ts'),
});
```
3. Get this result:
```

/** ------------------------------------------------------
 * THIS FILE WAS AUTOMATICALLY GENERATED (DO NOT MODIFY)
 * -------------------------------------------------------
 */

/* tslint:disable */
/* eslint-disable */
export interface CreateResponse {
    data?: string;
    query?: Query;
}

export interface IMutation {
    create(input: string): CreateResponse | Promise<CreateResponse>;
}

export interface IQuery {
    users(filter?: string): string | Promise<string>;
}
```

Line 11 of the result refers to `Query`, but the type name is `IQuery` on line 18.

## What is the motivation / use case for changing the behavior?
I'm trying to use a schema that was generated with Postgraphile, and the resulting TypeScript does not compile.

## Environment

<pre><code>
Nest version: 7.4.1
 
For Tooling issues:
- Node version: 12.13.0 
- Platform:  Windows 10

Others:
</code></pre>

**Comment by kamilmysliwiec** [comment]
Would you like to create a PR for this issue?

**Comment by kamilmysliwiec** [comment]
Fixed in 7.4.3
